app/controllers/application_controller.rb

```diff
@@ -69,28 +69,29 @@ def redirect_back_or_default(notice = '', default = root_url, alert = false)
 
   def get_manager_name(uniqname, program)
     result = {'valid' => false, 'note' => '', 'last_name' => '', 'first_name' => ''}
-    name = LdapLookup.get_simple_name(uniqname)
-    if name == "No such user"
-      result['note'] = "The '#{uniqname}' uniqname is not valid."
-    else
+    valid = LdapLookup.uid_exist?(uniqname)
```

Comment:
This code is a repeat of what you do in programs_controller. I am suggesting DRYing the code and making one method that can be called. 
You could also reduce the calls considerably by doing a uniqname lookup once when a uniqname is entered in the application. The caching of active record would then use that value across the session.

Reply:
The ldap_lookup gem is used in four different places (see the PR description). All these methods are slightly different. The code you mentioned in the application_controller is used when we add the manager to the program (in Programs::ManagersController on line 22.
In programs_controller we add an instructor in a program's new form.

> You could also reduce the calls considerably by doing a uniqname lookup once when a uniqname is entered in the application.
>
I don't understand what you mean here. A uniqname is entered in the application in four different places: an instructor's uniqname in a new program form, a manager's uniqname on a page to add managers to a program, a student's uniqname when we add students to a program, and an instructor's uniqname in a new program survey form.
How to reduce the number of calls?

> The caching of active record would then use that value across the session.
>
I don't understand the value of which active record you suggest to cache here.